File: frontend/lib/models/note.dart
class Note {
  final int id;
  final int userId;
  final String? content;
  final String? contentType;
  final String? summary;
  final String? category;
  final String? tags;
  final String? difficulty;
  final String? createTime;
  final String? updateTime;

  Note({
    required this.id,
    required this.userId,
    this.content,
    this.contentType,
    this.summary,
    this.category,
    this.tags,
    this.difficulty,
    this.createTime,
    this.updateTime,
  });

  factory Note.fromJson(Map<String, dynamic> json) {
    return Note(
      id: json['id'] ?? 0,
      userId: json['userId'] ?? 0,
      content: json['content'],
      contentType: json['contentType'],
      summary: json['summary'],
      category: json['category'],
      tags: json['tags'],
      difficulty: json['difficulty'],
      createTime: json['createTime'],
      updateTime: json['updateTime'],
    );
  }
}

class NoteDetail {
  final int id;
  final int userId;
  final String? content;
  final String? contentType;
  final String? summary;
  final String? category;
  final String? tags;
  final String? difficulty;
  final String? aiResult;
  final int? isReviewed;
  final String? createTime;
  final List<InterviewQuestion> interviewQuestions;

  NoteDetail({
    required this.id,
    required this.userId,
    this.content,
    this.contentType,
    this.summary,
    this.category,
    this.tags,
    this.difficulty,
    this.aiResult,
    this.isReviewed,
    this.createTime,
    this.interviewQuestions = const [],
  });

  factory NoteDetail.fromJson(Map<String, dynamic> json) {
    return NoteDetail(
      id: json['id'] ?? 0,
      userId: json['userId'] ?? 0,
      content: json['content'],
      contentType: json['contentType'],
      summary: json['summary'],
      category: json['category'],
      tags: json['tags'],
      difficulty: json['difficulty'],
      aiResult: json['aiResult'],
      isReviewed: json['isReviewed'],
      createTime: json['createTime'],
      interviewQuestions: (json['interviewQuestions'] as List<dynamic>?)
              ?.map((e) => InterviewQuestion.fromJson(e))
              .toList() ??
          [],
    );
  }
}

class InterviewQuestion {
  final int id;
  final String? question;
  final String? answer;
  final String? level;
  final int? isMastered;
  final String? createTime;

  InterviewQuestion({
    required this.id,
    this.question,
    this.answer,
    this.level,
    this.isMastered,
    this.createTime,
  });

  factory InterviewQuestion.fromJson(Map<String, dynamic> json) {
    return InterviewQuestion(
      id: json['id'] ?? 0,
      question: json['question'],
      answer: json['answer'],
      level: json['level'],
      isMastered: json['isMastered'],
      createTime: json['createTime'],
    );
  }
}

File: backend/src/main/java/com/mindflow/entity/Note.java
package com.mindflow.entity;

import com.baomidou.mybatisplus.annotation.*;
import java.time.LocalDateTime;

/**
 * 学习记录表
 */
@TableName("note")
public class Note {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String content;
    private String contentType;
    private String fileUrl;
    private String summary;
    private String category;
    private String tags;
    private String difficulty;
    private String aiResult;
    private Integer isReviewed;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;

    // ===== getters & setters =====
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
    public String getContentType() { return contentType; }
    public void setContentType(String contentType) { this.contentType = contentType; }
    public String getFileUrl() { return fileUrl; }
    public void setFileUrl(String fileUrl) { this.fileUrl = fileUrl; }
    public String getSummary() { return summary; }
    public void setSummary(String summary) { this.summary = summary; }
    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }
    public String getTags() { return tags; }
    public void setTags(String tags) { this.tags = tags; }
    public String getDifficulty() { return difficulty; }
    public void setDifficulty(String difficulty) { this.difficulty = difficulty; }
    public String getAiResult() { return aiResult; }
    public void setAiResult(String aiResult) { this.aiResult = aiResult; }
    public Integer getIsReviewed() { return isReviewed; }
    public void setIsReviewed(Integer isReviewed) { this.isReviewed = isReviewed; }
    public LocalDateTime getCreateTime() { return createTime; }
    public void setCreateTime(LocalDateTime createTime) { this.createTime = createTime; }
    public LocalDateTime getUpdateTime() { return updateTime; }
    public void setUpdateTime(LocalDateTime updateTime) { this.updateTime = updateTime; }
    public Integer getDeleted() { return deleted; }
    public void setDeleted(Integer deleted) { this.deleted = deleted; }
}